Fake 750Ti

multiflexi opened the pull request:
I recently flashed fake 750Ti. I would like to share how I did it. Pics and rom will follow.

multiflexi committed:
Fake 750Ti

I recently flashed fake 750Ti. I would like to share how I did it. I pics and rom will follow.

rigred commented:
You say that the GPU after flashing only has 768Mb?
Yet it has 8 2GBit RAM Chips? That means that it should have 2 Gbytes of VRAM not 768Mb
AM I understanding this correctly?

If so this means that the VRAM configuration of the BIOS you flashed is horribly wrong.

rigred commented:
I have a lot of details that I want to add.
The most important being about how to select an appropriate GPU BIOS that matches the hardware, RAM config, timings and display outputs.

A lot of people have been contacting me about it and it's the trickiest part with GPU flashing. 

I just wish I wasn't always tasked with other things so I had a lengthy bit of time to devote to this.

I will have a look at how I can merge your information.

multiflexi commented:
I will take pictures of PCB tomorrow. How should I share them with you?

rigred commented:
Simply attach them to the issue as file attachments. That would be the easiest, can pretty much just drag the file into the textbox.

multiflexi commented:
![clips-adapter](https://user-images.githubusercontent.com/11663844/37144305-ea999ff8-22bd-11e8-83bb-611a3c932c7a.jpg)
![fm25f02a](https://user-images.githubusercontent.com/11663844/37144306-eab69f72-22bd-11e8-8314-34cd6fc1acae.jpg)
![fake750ti-top](https://user-images.githubusercontent.com/11663844/37144307-ead3f39c-22bd-11e8-9fd5-eaf7386194d2.jpg)
![fake750ti-bottom](https://user-images.githubusercontent.com/11663844/37144308-eafb0e46-22bd-11e8-9b13-4a03bb706f23.jpg)
![fake750ti](https://user-images.githubusercontent.com/11663844/37144310-eb173576-22bd-11e8-850f-2bc12bf54b54.jpg)
![flashed](https://user-images.githubusercontent.com/11663844/37144311-eb3356de-22bd-11e8-8a83-f24ddd8538ef.png)
![ch341a](https://user-images.githubusercontent.com/11663844/37144312-eb50106c-22bd-11e8-9137-bee4aaacb080.jpg)
![fake750ti](https://user-images.githubusercontent.com/11663844/37144316-eb78df4c-22bd-11e8-9ed7-73c66657dd89.png)


[roms.zip](https://github.com/rigred/gpu-schmoo/files/1792589/roms.zip)

rigred commented:
Ok Nice.
I cant quite see the RAM markings fully. They are certainly 2Gbit 64Mx32 chips.
So in total the card should have 2Gbyte of VRAM as a lot of GTX550Ti's where configured.

Would you mind telling me which variant of this RAM you have?

| Speed | Model Number | Pinout | Bank | Voltage |  
| --------- | ------------------- | --------- | ------- | ---------- |
| 7.0Gbps | H5GQ2H24AFR-**R2C** | FBGA(170ball) | 16Bank | 1.6V/1.6V |
| 6.0Gbps | H5GQ2H24AFR-**R0C** | FBGA(170ball) | 16Bank | 1.5V/1.5V |
| 5.0Gbps | H5GQ2H24AFR-**T2C** | FBGA(170ball) | 16Bank | 1.5V/1.5V |

I'm assuming they are the T2C?

rigred commented:
Also run lspci with a few extra flags:

```
sudo lspci -vvv -knn -s 02:00
```

This should dump more lspci info.

Then if you can get yourself this repo:

https://github.com/envytools/envytools

Build it according to the instructions

And then run the `nvbios` tool against the original bios on your machine. It should produce a large output that you can dump into a text file. 

Then run it against the bios you flashed aswel.

multiflexi commented:
Yes, T2C.
Here is the lspci dump:

01:00.0 VGA compatible controller [0300]: NVIDIA Corporation GF116 [GeForce GTX 550 Ti] [10de:1244] (rev a1) (prog-if 00 [VGA controller])
	Subsystem: Micro-Star International Co., Ltd. [MSI] GF116 [GeForce GTX 550 Ti] [1462:809d]
	Control: I/O+ Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx-
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 0
	Interrupt: pin A routed to IRQ 125
	Region 0: Memory at da000000 (32-bit, non-prefetchable) [size=32M]
	Region 1: Memory at c0000000 (64-bit, prefetchable) [size=128M]
	Region 3: Memory at c8000000 (64-bit, prefetchable) [size=64M]
	Region 5: I/O ports at e000 [size=128]
	[virtual] Expansion ROM at 000c0000 [disabled] [size=128K]
	Capabilities: [60] Power Management version 3
		Flags: PMEClk- DSI- D1- D2- AuxCurrent=0mA PME(D0-,D1-,D2-,D3hot-,D3cold-)
		Status: D0 NoSoftRst+ PME-Enable- DSel=0 DScale=0 PME-
	Capabilities: [68] MSI: Enable+ Count=1/1 Maskable- 64bit+
		Address: 00000000fee40004  Data: 4022
	Capabilities: [78] Express (v2) Endpoint, MSI 00
		DevCap:	MaxPayload 128 bytes, PhantFunc 0, Latency L0s <4us, L1 <64us
			ExtTag+ AttnBtn- AttnInd- PwrInd- RBE+ FLReset- SlotPowerLimit 75.000W
		DevCtl:	Report errors: Correctable- Non-Fatal- Fatal- Unsupported-
			RlxdOrd+ ExtTag+ PhantFunc- AuxPwr- NoSnoop-
			MaxPayload 128 bytes, MaxReadReq 512 bytes
		DevSta:	CorrErr- UncorrErr- FatalErr- UnsuppReq- AuxPwr- TransPend-
		LnkCap:	Port #0, Speed 2.5GT/s, Width x16, ASPM L0s L1, Exit Latency L0s <256ns, L1 <4us
			ClockPM+ Surprise- LLActRep- BwNot- ASPMOptComp-
		LnkCtl:	ASPM Disabled; RCB 64 bytes Disabled- CommClk+
			ExtSynch- ClockPM+ AutWidDis- BWInt- AutBWInt-
		LnkSta:	Speed 2.5GT/s, Width x16, TrErr- Train- SlotClk+ DLActive- BWMgmt- ABWMgmt-
		DevCap2: Completion Timeout: Not Supported, TimeoutDis+, LTR-, OBFF Not Supported
		DevCtl2: Completion Timeout: 50us to 50ms, TimeoutDis-, LTR-, OBFF Disabled
		LnkCtl2: Target Link Speed: 2.5GT/s, EnterCompliance- SpeedDis-
			 Transmit Margin: Normal Operating Range, EnterModifiedCompliance- ComplianceSOS-
			 Compliance De-emphasis: -6dB
		LnkSta2: Current De-emphasis Level: -6dB, EqualizationComplete-, EqualizationPhase1-
			 EqualizationPhase2-, EqualizationPhase3-, LinkEqualizationRequest-
	Capabilities: [b4] Vendor Specific Information: Len=14 <?>
	Capabilities: [100 v1] Virtual Channel
		Caps:	LPEVC=0 RefClk=100ns PATEntryBits=1
		Arb:	Fixed- WRR32- WRR64- WRR128-
		Ctrl:	ArbSelect=Fixed
		Status:	InProgress-
		VC0:	Caps:	PATOffset=00 MaxTimeSlots=1 RejSnoopTrans-
			Arb:	Fixed- WRR32- WRR64- WRR128- TWRR128- WRR256-
			Ctrl:	Enable+ ID=0 ArbSelect=Fixed TC/VC=ff
			Status:	NegoPending- InProgress-
	Capabilities: [128 v1] Power Budgeting <?>
	Capabilities: [600 v1] Vendor Specific Information: ID=0001 Rev=1 Len=024 <?>
	Kernel driver in use: nvidia
	Kernel modules: nvidiafb, nouveau, nvidia_drm, nvidia

01:00.1 Audio device [0403]: NVIDIA Corporation GF116 High Definition Audio Controller [10de:0bee] (rev a1)
	Subsystem: Micro-Star International Co., Ltd. [MSI] GF116 High Definition Audio Controller [1462:809d]
	Control: I/O- Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx-
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 0
	Interrupt: pin B routed to IRQ 17
	Region 0: Memory at dc080000 (32-bit, non-prefetchable) [size=16K]
	Capabilities: [60] Power Management version 3
		Flags: PMEClk- DSI- D1- D2- AuxCurrent=0mA PME(D0-,D1-,D2-,D3hot-,D3cold-)
		Status: D0 NoSoftRst+ PME-Enable- DSel=0 DScale=0 PME-
	Capabilities: [68] MSI: Enable- Count=1/1 Maskable- 64bit+
		Address: 0000000000000000  Data: 0000
	Capabilities: [78] Express (v2) Endpoint, MSI 00
		DevCap:	MaxPayload 128 bytes, PhantFunc 0, Latency L0s <4us, L1 <64us
			ExtTag+ AttnBtn- AttnInd- PwrInd- RBE+ FLReset- SlotPowerLimit 75.000W
		DevCtl:	Report errors: Correctable- Non-Fatal- Fatal- Unsupported-
			RlxdOrd+ ExtTag+ PhantFunc- AuxPwr- NoSnoop+
			MaxPayload 128 bytes, MaxReadReq 512 bytes
		DevSta:	CorrErr+ UncorrErr- FatalErr- UnsuppReq- AuxPwr- TransPend-
		LnkCap:	Port #0, Speed 2.5GT/s, Width x16, ASPM L0s L1, Exit Latency L0s <256ns, L1 <4us
			ClockPM+ Surprise- LLActRep- BwNot- ASPMOptComp-
		LnkCtl:	ASPM L0s L1 Enabled; RCB 128 bytes Disabled- CommClk+
			ExtSynch- ClockPM+ AutWidDis- BWInt- AutBWInt-
		LnkSta:	Speed 2.5GT/s, Width x16, TrErr- Train- SlotClk+ DLActive- BWMgmt- ABWMgmt-
		DevCap2: Completion Timeout: Not Supported, TimeoutDis+, LTR-, OBFF Not Supported
		DevCtl2: Completion Timeout: 50us to 50ms, TimeoutDis-, LTR-, OBFF Disabled
		LnkSta2: Current De-emphasis Level: -6dB, EqualizationComplete-, EqualizationPhase1-
			 EqualizationPhase2-, EqualizationPhase3-, LinkEqualizationRequest-
	Kernel driver in use: snd_hda_intel
	Kernel modules: snd_hda_intel


And output from nvbios

[fakerom.zip](https://github.com/rigred/gpu-schmoo/files/1793504/fakerom.zip)

[550tirom.txt](https://github.com/rigred/gpu-schmoo/files/1793508/550tirom.txt)
